src/protocol/codec/transport.c

```diff
@@ -388,25 +434,53 @@ z_result_t _z_fragment_encode(_z_wbuf_t *wbf, uint8_t header, const _z_t_msg_fra
     z_result_t ret = _Z_RES_OK;
     _Z_DEBUG("Encoding _Z_TRANSPORT_FRAGMENT");
     _Z_RETURN_IF_ERR(_z_zsize_encode(wbf, msg->_sn))
-    if (_Z_HAS_FLAG(header, _Z_FLAG_T_Z)) {
-        ret = _Z_ERR_MESSAGE_SERIALIZATION_FAILED;
+    if (msg->first) {
+        if (_Z_HAS_FLAG(header, _Z_FLAG_T_Z) == true) {
```

Comment:
Just a nit and I see there are some that precedes that PR, but all conditions should be booleans so the `== true / false` would be redundant and can be removed.

Reply:
See https://github.com/eclipse-zenoh/zenoh-pico/pull/813#discussion_r1868961287

Reply:
@sashacmc @jean-roland @wyfo this is about MISRA C rules: https://stackoverflow.com/questions/44131637/misra-c-2012-rule-10-1-boolean-operand-to-be-used-in-case-where-expression-is-of

Reply:
AFAIK, MISRA-C rules about using non-boolean values as boolean expression:
e. g.
```
int i = 1;
if (i) {}
```
but this kind of expression is absolutely legal:
```
bool b = true;
if (b) {}
```

Reply:
Well, yes but I also believe it depends on the static analyser... I'm not saying we should use `== true`, I was just providing a bit more context :)